prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -18,135 +21,162 @@
 from prediction_market_agent.tools.custom_crewai_tools import TavilyDevTool
 from prediction_market_agent.utils import APIKeys
 
-tavily_search = TavilyDevTool()
 
-
-class Outcomes(BaseModel):
-    outcomes: list[str]
+class Scenarios(BaseModel):
+    scenarios: list[str]
 
 
 class ProbabilityOutput(BaseModel):
-    decision: str
+    reasoning: str
+    decision: t.Literal["y", "n"] | None
     p_yes: float
     p_no: float
     confidence: float
 
 
 class CrewAIAgentSubquestions:
-    def __init__(self) -> None:
-        llm = self._build_llm()
-        self.researcher = Agent(
+    model: str = "gpt-4-turbo-2024-04-09"
+
+    def _get_current_date(self) -> str:
+        return utcnow().strftime("%Y-%m-%d")
+
+    def _get_researcher(self) -> Agent:
```

Comment:
I added multi-threading for predicting individual subquestions. They are independent of each other so it speeds it up a lot. But because of that, every prediction needs to create its own agent instances.

Reply:
Left a comment on parallelism